site-packages/integralstor/cifs.py

```diff
@@ -531,18 +531,18 @@ def generate_smb_conf():
         return True, None
 
 
-def reload_configuration():
+def reload_configuration(action='restart'):
     try:
         cmd_to_run = 'smbcontrol all reload-config'
```

Comment:
Can we have a check here to make sure that the values for action are only either reload or restart and if not, it can throw an exception with the appropriate error string.

Reply:
Okay, shall I add this block here:

```
if action not in ['reload', 'restart']:
    raise exception("Received Invalid argument")
```